From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Tony Hutter <hutter2@llnl.gov>
Date: Fri, 26 Oct 2018 06:51:38 -0400
Subject: [PATCH] Backport vnode.h changes from zfs:93ce2b4c

This backports the vnode.h changes from zfs:93ce2b4c for zfs-0.7.12.

Signed-off-by: Tony Hutter <hutter2@llnl.gov>
Signed-off-by: Stoiko Ivanov <s.ivanov@proxmox.com>
---
 include/sys/vnode.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/include/sys/vnode.h b/include/sys/vnode.h
index 87f12d6..279bd87 100644
--- a/include/sys/vnode.h
+++ b/include/sys/vnode.h
@@ -87,7 +87,7 @@
 #define	AT_MTIME	ATTR_MTIME
 #define	AT_CTIME	ATTR_CTIME
 
-#define	ATTR_XVATTR	(1 << 31)
+#define	ATTR_XVATTR	(1U << 31)
 #define	AT_XVATTR	ATTR_XVATTR
 
 #define	ATTR_IATTR_MASK	(ATTR_MODE | ATTR_UID | ATTR_GID | ATTR_SIZE | \
@@ -121,7 +121,7 @@ typedef enum vtype {
 
 typedef struct vattr {
 	enum vtype	va_type;	/* vnode type */
-	uint_t		va_mask;	/* attribute bit-mask */
+	uint32_t        va_mask;        /* attribute bit-mask */
 	ushort_t	va_mode;	/* acc mode */
 	uid_t		va_uid;		/* owner uid */
 	gid_t		va_gid;		/* owner gid */
